server/expression/executor_test.go

```diff
@@ -373,6 +373,48 @@ func TestArrayExecution(t *testing.T) {
 			Query:      `[31,35,39] contains 42`,
 			ShouldPass: false,
 		},
+		{
+			Name:       "should_identify_array_instead_of_json",
+			Query:      `["{}", "{}", "{}"] | type = "array"`,
+			ShouldPass: true,
+		},
+	}
+
+	executeTestCases(t, testCases)
+}
+
+func TestJSONExecution(t *testing.T) {
```

Comment:
Does it make sense to add a comparison for escaped JSON? Something like:
```
'{"name": "john", "age": 32, "email": "john@company.com"}' contains '{\"email\": \"john@company.com\" }'
```

Reply:
Escaping is not stable in the engine. As this is a problem in the core of the engine and not in the feature, I'll skip this and wait for users to ask for a fix. Not sure if that will happen.